### PR TITLE
[HTTPDB] Replace invalid utf-8 characters on watch logs

### PR DESCRIPTION
**Pull request by alonmr**
https://jira.iguazeng.com/browse/ML-3653

### CODE REVIEW

File: tests/rundb/test_httpdb.py [hunk 1]
```diff
@@ -747,3 +750,65 @@ def _assert_projects(expected_project, project):
...
+        content=callback,
+    )
+    db.session = db._init_session()
+    db.session.mount("http+mock", adapter)
+    mlrun.mlconf.httpdb.logs.pull_logs_default_interval = 0.1
+    with unittest.mock.patch("sys.stdout", new_callable=io.StringIO) as newprint:
+        db.watch_log(run_uid, project=project)
+        assert (
+            newprint.getvalue()
+            == log_contents[0 : len(log_lines[0])].decode(errors="replace")
```
Comment (theSaarco) [inline tests/rundb/test_httpdb.py:806]:
```suggestion
            == log_contents[0 : len(log_lines[0])].decode(errors=mlrun.mlconf.httpdb.logs.decode.errors)
```
Unless we assume the config mode will always be `replace` in which case there's no point in putting it in a config parameter.

**Review by theSaarco — CHANGES_REQUESTED** [review]
Looks good - I have questions/comments on the test.

File: tests/rundb/test_httpdb.py [hunk 2]
```diff
@@ -747,3 +750,65 @@ def _assert_projects(expected_project, project):
...
+            + "\n"  # the first log line is printed with a newline
```
[the remark this answers is not in the document]
Comment (alonmr) [reply tests/rundb/test_httpdb.py:807]: https://github.com/mlrun/mlrun/blob/development/mlrun/db/httpdb.py#L460
I'm not sure if it's a bug or not but if we want to remove it we need to use `print(... ,end="")`

File: tests/rundb/test_httpdb.py [hunk 3]
```diff
@@ -747,3 +750,65 @@ def _assert_projects(expected_project, project):
...
+            + log_contents[len(log_lines[0]) :].decode(errors="replace")
+            == "Firstrow\nSecondrowThirdrowSmiley😆�LastRow"
```
[the remark this answers is not in the document]
Comment (liranbg) [reply tests/rundb/test_httpdb.py:809]: human is enough imo @alonmr

**Review by liranbg — APPROVED** [review]

**Review by theSaarco — APPROVED** [review]
LGTM.
One note I would consider before merging - since the test you added assumes the decode-mode is `replace` (the assertion assumes it), suggest to start the test with `mlrun.mlconf.httpdb.logs.decode.errors = "replace"`, just to make sure it's stable even if the decode mode is changed for some reason.

**Pull request merged**